zstash/ls.py

```diff
@@ -1,12 +1,17 @@
-from __future__ import print_function, absolute_import
+from __future__ import absolute_import, print_function
 
-import os
-import sys
 import argparse
 import logging
+import os
 import sqlite3
+import sys
+from typing import List
+
 from .hpss import hpss_get
-from .settings import config, logger, BLOCK_SIZE, DEFAULT_CACHE, get_db_filename
+from .settings import DEFAULT_CACHE, config, get_db_filename, logger
+
+con = None
+cur = None
```

Comment:
>global tells python to look for a variable with this name in the global namespace and include it in the local namespace. This means it must exist in the global namespace first.
https://stackoverflow.com/a/55765437

Reply:
I'm not sure why those were even defined as `global`. There's not even a second function in this file to make use of a global variable.

Reply:
Yeah, it looks like `con` and `cur` are only used in one function for each file that it exists in (`create.py`, `extract.py`, `ls.py`, and `update.py`).

```
   global con, cur
    con = sqlite3.connect(get_db_filename(cache), detect_types=sqlite3.PARSE_DECLTYPES)
    cur = con.cursor()
```

I noticed some repeated logic involving database work. It might be better to declare the `con` and `cur` variables in `main.py` and pass them into each of the functions, rather than instantiating a new `con` and `cur` variables in each function.

https://github.com/E3SM-Project/zstash/blob/master/zstash/main.py#L51
```
    if args.command == 'version':
        print(__version__)
    elif args.command == 'create':
        create()
    elif args.command == 'update':
        update()
    elif args.command == 'extract':
        extract()
    elif args.command == 'chgrp':
        chgrp()
    elif args.command == 'check':
        check()
    elif args.command == 'ls':
        ls()
    else:
        print('Unrecognized command')
        parser.print_help()
        sys.exit(1)
```